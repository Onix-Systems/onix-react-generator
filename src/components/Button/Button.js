import React from 'react';
import PropTypes from 'prop-types';

export default function Button({ title, buttonClass, handleClick }) {
  return (
    <button
      type="button"
      onClick={handleClick}
      className={buttonClass}
      data-testid="button"
    >
      { title }
    </button>
  );
}

Button.propTypes = {
  title: PropTypes.string.isRequired,
  buttonClass: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired
};
