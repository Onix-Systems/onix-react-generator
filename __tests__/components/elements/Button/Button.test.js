import React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from '../../../settings/test-utils';
import Button from '../../../../src/components/Button/Button';

describe('<Button />', () => {
  let wrapper;
  const mockButtonClass = 'mock button class';
  const mockHandleClick = jest.fn();
  const mockButtonTitle = 'mock title';

  beforeEach(() => {
    wrapper = render(
      <Button
        title={mockButtonTitle}
        buttonClass={mockButtonClass}
        handleClick={mockHandleClick}
      />
    );
  });
  test('should render Button component', () => {
    const { asFragment } = wrapper;
    expect(asFragment(wrapper)).toMatchSnapshot();
  });

  test('should call handleClick function on click', () => {
    const { getByText } = wrapper;
    const button = getByText(mockButtonTitle);
    userEvent.click(button);
    expect(mockHandleClick).toHaveBeenCalled();
  });
});
