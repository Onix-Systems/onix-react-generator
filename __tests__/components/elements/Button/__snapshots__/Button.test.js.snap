// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render Button component 1`] = `
<DocumentFragment>
  <button
    class="mock button class"
    data-testid="button"
    type="button"
  >
    mock title
  </button>
</DocumentFragment>
`;
